contents/csharp/IntroductionToCSharp.md

```diff
@@ -0,0 +1,186 @@
+
+## Some Noteworthy Features of C#
+
+Most programming languages/frameworks have their own unique quirks, which can be utilised for better code quality.
+
+### Object / Array / Collection Initializers
+
+In C#, you can construct an [Object](https://en.wikipedia.org/wiki/Object_(computer_science) "In computer science, an object can be a variable, a data structure, a function, or a method, and as such, is a value in memory referenced by an identifier.")
+, [Array](https://www.webopedia.com/TERM/A/array.html "In programming, a series of objects all of which are the same size and type. Each object in an array is called an array element. For example, you could have an array of integers or an array of characters or an array of anything that has a defined data type.") 
+or [Collection](https://computersciencewiki.org/index.php/Collections "A collection — sometimes called a container — is simply an object that groups multiple elements into a single unit.") conveniently as shown. This can be useful when writing tests, as test data will be easier to read,
+as compared to calling the actual constructor.
+
+```csharp
+public class Foo {
+    Bar MyBar;
+}
+
+Bar bar = new Bar();
+Foo foo = new Foo() {
+    MyBar = bar;
+};
+```
+
+### Closures
+
+Sometimes, it may be useful to [defer execution](http://www.informit.com/articles/article.aspx?p=2171751 "Code that is executed only when results need to be evaluated. There are many reasons for executing code later") 
+or capture a local context for later execution. Context capturing is reflected below:
+
+```csharp
+//Capturing local context
+public class Context {
+    public static Func<Int> GetCounter() {
+        int count = 0;
+        return () => {
+            count++;
+            return count();
+        }
+    }
+}
+
+Func<Int> counter = Context.GetCounter();
+counter();
+```
+
+### Nullable type
+
+Often when dealing with possible sites of [null pointer](https://en.wikipedia.org/wiki/Null_pointer "In computing, a null pointer or null reference has a value reserved for indicating that the pointer or reference does not refer to a valid object.") 
+access, many if checks might be used. However, when dealing with nullable types, such code can be shortened
+while remaining easy to read, using collaescing operators `??`, or null conditional operators `.?`. Some may find this similar to `optionals` in Swift.
+
+More can be read about Nullables [here](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/)
+
+```csharp
+public Car? ManufactureCar() {
+    return (hasError) ? null : new Car(param1, param2);
+}
+
+Car? car = ManufactureCar();
+//Call Drive method of car if not null
+car?.Drive();
+
+//Get amount of fuel left
+Double fuelLeft = car?.GetFuel() ?? 0;
+//Explanation:
+//If car is not null, one can expect the statement to evaluate to car.GetFuel().
+//If car is null, car?.GetFuel() evaluates to null
+//?? operator sees null, so the entire expression defaults to 0.
+```
```

Comment:
This point too can benefit from more detailed explanations and code to compare _with_ and _without_ the feature.